code/modules/clothing/spacesuits/rig/vault_hardsuit_token.dm

```diff
@@ -0,0 +1,34 @@
+/obj/item/vault_token
+	name = "hardsuit token"
+	desc = "A token infused with telecrystals. You can summon almost any hardsuit with this."
+	icon = 'icons/obj/vault_token.dmi'
+	icon_state = "hardsuit_token"
+
+/obj/item/vault_token/attack_self(mob/living/carbon/human/user)
+	if(!ishuman(user))
+		to_chat(user, SPAN_WARNING("You can't use \the [src]."))
+		return
+	var/list/hardsuit_options = list(
+		"Crimson Hardsuit" = /obj/structure/closet/crate/secure/gear_loadout/vault/crimson,
+		"Solarian Military Hardsuit" = /obj/structure/closet/crate/secure/gear_loadout/vault/sol,
+		"Hazard Control Hardsuit" = /obj/structure/closet/crate/secure/gear_loadout/vault/hazard,
+		"Einstein Hardsuit" = /obj/structure/closet/crate/secure/gear_loadout/vault/einstein,
+		"Combat Hardsuit" = /obj/structure/closet/crate/secure/gear_loadout/vault/combat
+	)
+	for(var/hardsuit_option in hardsuit_options)
+		var/crate_path = hardsuit_options[hardsuit_option]
+		var/obj/structure/closet/crate/secure/gear_loadout/vault/N = crate_path
+		var/hardsuit_path = initial(N.associated_vault_hardsuit)
+		var/obj/item/rig/R = new hardsuit_path
+		if(!(user.species.bodytype in R.species_restricted))
+			hardsuit_options -= hardsuit_option
+	var/list/options = list()
+	for(var/hardsuit in hardsuit_options)
+		var/image/radial_button = image(icon = src.icon, icon_state = hardsuit)
```

Comment:
You should probably use the hardsuit icons from the file itself by building a mannequin rather than hardcoding them in.

Reply:
I copied existing code and edited it just enough to work here. The radial menu draws from the vault_token.dmi. 
I won't lie, I don't even know how to build a mannequin. I just know that this works and is used for the Ninja token, so.